utilities/ideascale-importer/ideascale_importer/snapshot_importer/importer.py

```diff
@@ -115,11 +118,23 @@ async def _fetch_parameters(self):
 
             self.snapshot_tool_max_slot = row["slot_no"]
             logger.info(
-                "Got max_slot for max_time",
+                "Got block time parameters for snapshot_tool",
                 snapshot_tool_max_slot=self.snapshot_tool_max_slot,
                 max_time=self.config.snapshot_tool.max_time.isoformat()
             )
 
+            row = await conn.fetchrow(
+                "SELECT max(time) AS latest_time FROM block",
```

Comment:
So we should get the latest slot_no here, just in case the query above failed, and so we can use it as the slot_no we run the snapshot against.  The as_at in the db would then need to be the time from this query if it defaults here.